packages/app/src/cli/services/generate/fetch-extension-specifications.ts

```diff
@@ -56,20 +61,26 @@ async function mergeLocalAndRemoteSpecs(
   local: ExtensionSpecification[],
   remote: FlattenedRemoteSpecification[],
 ): Promise<RemoteAwareExtensionSpecification[]> {
-  const updated = local.map(async (spec) => {
-    const remoteSpec = remote.find((remote) => remote.identifier === spec.identifier)
-    if (remoteSpec) {
-      const merged = {...spec, ...remoteSpec, loadedRemoteSpecs: true} as RemoteAwareExtensionSpecification &
-        FlattenedRemoteSpecification
+  // Iterate over the remote specs and merge them with the local ones
+  // If the local spec is missing, and the remote one has a validation schema, create a new local spec using contracts
+  const updated = remote.map(async (remoteSpec) => {
+    let localSpec = local.find((local) => local.identifier === remoteSpec.identifier)
+    if (!localSpec && remoteSpec.validationSchema?.jsonSchema) {
+      const normalisedSchema = await normaliseJsonSchema(remoteSpec.validationSchema.jsonSchema)
+      const hasLocalization = normalisedSchema.properties?.localization !== undefined
```

Comment:
Should we be porting this to something on the specifications API instead? The CLI feature list in general I suppose. Should we block this on updating the API or proceed without?